python/lsst/ts/eas/dome_model.py

```diff
@@ -0,0 +1,99 @@
+# This file is part of ts_eas.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["DomeModel"]
+
+import asyncio
+
+from lsst.ts import salobj, utils
+
+DORMANT_TIME = 300  # Time to wait while sleeping, seconds
+MAX_TELEMETRY_AGE = 300  # Time at which apertureShutter telemetry expires, seconds
+
+
+class DomeModel:
+    """A model for MTDome.
```

Comment:
Why not base this on a set of events? You pass the dome model. The thing that needs to wait for dome event creates asyncio.event and add that's to dome model on_open list. Dome, when it detects the dome opened/closed, fires the events, and remove those from the list. Waiting things ends waits, clears the events, and might ask again to sleep until dome is opened/closed. The busy waits works, this gets you more control over what's handled and what's not.